pytest_fixtures/core/sat_cap_factory.py

```diff
@@ -387,7 +387,7 @@ def installer_satellite(request):
         configure_nailgun()
         configure_airgun()
     yield sat
-    if 'sanity' not in request.config.option.markexpr:
+    if 'sanity' not in request.config.option.markexpr and settings.server.auto_checkin:
```

Comment:
@omkarkhatavkar If I'm reading this correctly, the checkin won't happen for the sanity server anyway.

Reply:
May I confuse you here, I'm talking here or this PR not want to check out the build_sanity satellite.

Reply:
If we run pytest with `-m build_sanity`, then `'sanity' not in request.config.option.markexpr` evaluates to `False`, therefore if `settings.server.auto_checkin` is set to `False` for the build sanity run, then we get
```python
if False and False
```
which evaluates to `False`. Therefore the following logical block will not get executed.

In a scenario where we run standard test automation, it would be `if True and True`, therefore the `sat` instance would be checked in unless we change the `auto_checkin` setting to False.

I see no issue with the logic Omkar proposes. Are there any problems that might arise if this patch goes in, @jameerpathan111?

Reply:
Anyway, like you said I don't see any problems with having this patch either.

Reply:
I did not state it explicitly in the above comment, but @omkarkhatavkar, could you please revisit the logic? For sanity testing with `auto_checkin` set to True, the condition would be `if False and True` -> `False`. Jameer is correct in what he is saying, there is no problem to solve, `installer_satellite` is not being checked in in case of sanity testing.

If we want to do a VM check-in in robottelo, I would prefer to relo solely on the `auto_checkin` setting. So the condition would look like
```python
if  settings.server.auto_checkin:
    do_checkin()
```